src/averageAges.js

```diff
@@ -15,13 +19,26 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people
+    .filter((person) => person.sex === SEX_MALE
+     && (century ? getCentry(person.died) === century : true)
+    );
+
+  return calculateAverageAge(men);
+}
+
+function calculateAverageAge(people) {
+  return people
+    .reduce((acc, person) => acc + getAge(person), 0) / people.length;
 }
 
+function getAge(person) {
+  return person.died - person.born;
+}
+
+function getCentry(deathYear) {
+  return Math.ceil(deathYear / CENTURY_VALUE);
+}
```

Comment:
Usually, such functions that are used throughout the code as helpers are kept together either after all logical functions or before all logical functions.